web/src/types/DAG.types.ts

```diff
@@ -0,0 +1,18 @@
+import {NodeTypesEnum} from 'constants/DAG.constants';
+import {SemanticGroupNames} from 'constants/SemanticGroupNames.constants';
+
+export interface INodeDatum<T> {
+  data: T;
+  id: string;
+  parentIds: string[];
+  type: NodeTypesEnum;
+}
+
+export interface INodeDataSpan {
```

Comment:
And this is the `data` interface for our DAG. This will change once we start adding more fields/info to the graph.